allennlp/data/dataset_readers/wikitables/type_declaration.py

```diff
@@ -0,0 +1,73 @@
+"""
+Defines all the types in the WikitablesQuestions domain. We exploit the type logic in ``nltk.sem.logic``
+here. This module also contains two helper classes that add some functionality on top of NLTK's logic module.
+"""
+from overrides import overrides
+
+from nltk.sem.logic import BasicType, ComplexType, EntityType, ANY_TYPE
+
+
+class BasicTypeFactory(BasicType):
```

Comment:
If you really want to do something like this, I think it's sufficient to just do this:

```python
class NamedBasicType(BasicType):
  def __init__(self, name):
    self.name = name
  def __str__(self) -> str:
    return self.name.lower()
```
And then you call `ROW_TYPE = NamedBasicType("ROW")` below.

But, given that you only do this twice, I think it'd just be cleaner to create two actual subclasses of `BasicType`, `RowType` and `DateNumType`, and then you'd match the nltk API better.

Reply:
Agreed. Changed it to `NamedBasicType` for now. I may wither split number and date or define a hierarchy for them later.